project-base/storefront/components/Pages/Order/PaymentConfirmation/Gateways/GoPayGateway.tsx

```diff
@@ -1,21 +1,29 @@
-import { Link } from 'components/Basic/Link/Link';
+import { SpinnerIcon } from 'components/Basic/Icon/IconsSvg';
+import { Button } from 'components/Forms/Button/Button';
 import { GoPayCreatePaymentSetupApi, usePayOrderMutationApi } from 'graphql/generated';
 import { showErrorMessage } from 'helpers/toasts';
 import useTranslation from 'next-translate/useTranslation';
 import Script from 'next/script';
-import { useEffect, useState } from 'react';
+import { useEffect, useRef, useState } from 'react';
 
 type GoPayGatewayProps = {
     orderUuid: string;
+    requiresAction?: boolean;
+    initialButtonText?: string;
 };
 
-export const GoPayGateway: FC<GoPayGatewayProps> = ({ orderUuid }) => {
-    const { t } = useTranslation();
-    const [isRedirectLinkVisible, setRedirectLinkVisibility] = useState(false);
+export const GoPayGateway: FC<GoPayGatewayProps> = ({ orderUuid, requiresAction, initialButtonText, className }) => {
+    const [initiatedPaymentGate, setInitiatedPaymentGate] = useState(!requiresAction);
     const [goPayPaymentSetup, setGoPayPaymentSetup] = useState<GoPayCreatePaymentSetupApi | undefined>(undefined);
     const [, payOrder] = usePayOrderMutationApi();
+    const { t } = useTranslation();
+    const wasPaidRef = useRef(false);
 
     useEffect(() => {
+        if (wasPaidRef.current || !initiatedPaymentGate) {
+            return;
+        }
+
         payOrder({ orderUuid }).then((payOrderResult) => {
```

Comment:
What will happen in case `payOrder` will throw an error? Don't we want to handle those cases as well?

Reply:
I am not sure what you mean. The error is handled right below. If you are asking about the `.catch` clause, then from what I understand, the mutation function cannot directly throw. If they do, it is not a request error.

<img width="734" alt="image" src="https://github.com/shopsys/shopsys/assets/55181354/ceac42f7-0013-40fd-a389-1476b6fe7f3a">